src/tests/test_process.py

```diff
@@ -32,10 +55,25 @@ class TestProcessProto(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
         """Setup environment and start pybitmessage"""
-        cls.home = os.environ['BITMESSAGE_HOME'] = tempfile.gettempdir()
+        cls.home = set_appdata_folder(tempfile.gettempdir())
+        cls._cleanup_files()
         put_signal_file(cls.home, 'unittest.lock')
+        starttime = time.time()
         subprocess.call(cls._process_cmd)  # nosec
-        time.sleep(5)
+        timeout = starttime + 30.0
+        while time.time() <= timeout:
+            try:
+                pstat = os.stat(os.path.join(cls.home, 'singleton.lock'))
+                if starttime <= pstat.st_mtime and pstat.st_size > 0:
+                    break
+            except OSError:
+                pass
+            time.sleep(1)
```

Comment:
Not so easy: https://travis-ci.org/github/g1itch/PyBitmessage/builds/738699572

Reply:
Rebased in my favor: https://travis-ci.org/github/g1itch/PyBitmessage/builds/738705081

Reply:
I added `os.path.exists`.

Reply:
if not `os.path.exists` it will be full wait until `pybitmessage -t` stops. See:
https://github.com/Bitmessage/PyBitmessage/blob/v0.6/src/bitmessagemain.py#L180
https://github.com/Bitmessage/PyBitmessage/blob/v0.6/src/bitmessagemain.py#L371

Reply:
Hmm https://travis-ci.org/github/g1itch/PyBitmessage/jobs/739258079

Reply:
no fractional part in `st_mtime`: https://travis-ci.org/github/g1itch/PyBitmessage/jobs/739262652

Reply:
Ok I'll look throught this.